### ISSUE TITLE
API: Make reset() private and call it from join() and personalize()

### ISSUE THREAD

**Issue by FokkeZB** [TheThingsNetwork/arduino-device-lib#6]
All examples call `ttn.reset()` before they call `ttn.join()` or `ttn.personalize()`. If this is a required step, why not make it private and call it from these two methods.

Current:

```c
ttn.reset();
ttn.personalize(devAddr, nwkSKey, appSKey);
```

Result:

```c
ttn.personalize(devAddr, nwkSKey, appSKey);
```

To @johanstokking's note, if the device comes pre-flashed with keys you'd call the methods with no arguments:

```c
while (ttn.join()) {
}
```

The methods should then not call reset() and skip the `mac set` commands.

**Comment by johanstokking** [comment]
Yes, I think we can do the `reset()` in `personalize()` and `join()`. The reason why it's separate, was to be able not to reset and join with the values that are stored in the RN module, instead of providing the values. But we can account for that in `join()` and/or with an overload that doesn't take keys and therefore doesn't reset.

**Comment by FokkeZB** [comment]
Yes, that seems like a good simplification since doing a `reset()` or `personalize()` without `reset()` will be the common scenario. I'll leave this assigned to you as I'm finding my way around in C/Arduino still ;)

**Comment by FokkeZB** [comment]
Of course if we call `reset()` from `join()` this will mean that if it takes a while to do a successful join, every new attempt will first `reset()` the device. This is a bit of an overhead.

**Comment by johanstokking** [comment]
Also, I think we should make it optional not to reset the device at all. I can imagine the AppEUI and AppKey being pre-flashed, especially in The Things Node, that we don't want to reset.

**Comment by FokkeZB** [comment]
Closed by #58
